lib/native/mqtt.ts

```diff
@@ -627,6 +629,18 @@ export class MqttClientConnection extends NativeResourceMixin(BufferedEventEmitt
         };
     }
 
+    private _on_connection_failure(error_code: number) {
+        let failureCallbackData = { error: new CrtError(error_code) } as OnConnectionFailedResult;
+        this.emit('connection_failure', failureCallbackData);
+
+    }
+    private _on_connection_success(return_code: number, session_present: boolean) {
+
+        let successCallbackData = { session_present: session_present, reason_code: return_code } as OnConnectionSuccessResult;
+        this.emit('connection_success', successCallbackData);
+
+    }
```

Comment:
If I'm not mistaken, in some cases both `_on_connection_resumed` and `_on_connection_success` will be called together:
https://github.com/awslabs/aws-c-mqtt/blob/main/source/client_channel_handler.c#L242
https://github.com/awslabs/aws-c-mqtt/blob/main/source/client_channel_handler.c#L263

So, `this.emit('connection_success', successCallbackData);` should be removed from `_on_connection_resumed `. Probably other functions (e.g. `_on_connect_callback`) should be checked as well.

I couldn't find on_resume callback usage in tests (it's tricky to test it on this layer), but `_on_connect_callback` should be called on every successful connect. So, as an idea, you could add a check to the tests that `on_connection_success`/`on_connection_failure` are fired exactly once.